tests/Aspire.Azure.AI.OpenAI.Tests/AspireAzureAIOpenAIExtensionsTests.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.AI.OpenAI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace Aspire.Azure.AI.OpenAI.Tests;
+
+public class AspireAzureAIOpenAIExtensionsTests
+{
+    private const string ConnectionString = "Endpoint=https://aspireopenaitests.openai.azure.com/;Key=fake";
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ReadsFromConnectionStringsCorrectly(bool useKeyed)
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>("ConnectionStrings:openai", ConnectionString)
+        ]);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedAzureOpenAI("openai");
+        }
+        else
+        {
+            builder.AddAzureOpenAI("openai");
+        }
+
+        var host = builder.Build();
+        var client = useKeyed ?
+            host.Services.GetRequiredKeyedService<OpenAIClient>("openai") :
+            host.Services.GetRequiredService<OpenAIClient>();
+
+        Assert.NotNull(client);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ConnectionStringCanBeSetInCode(bool useKeyed)
+    {
+        var uri = new Uri("https://aspireopenaitests.openai.azure.com/");
+        var key = "fake";
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedAzureOpenAI("openai", settings => { settings.Endpoint = uri; settings.Key = key; });
+        }
+        else
+        {
+            builder.AddAzureOpenAI("openai", settings => { settings.Endpoint = uri; settings.Key = key; });
+        }
+
+        var host = builder.Build();
+        var client = useKeyed ?
+            host.Services.GetRequiredKeyedService<OpenAIClient>("openai") :
+            host.Services.GetRequiredService<OpenAIClient>();
+
+        Assert.NotNull(client);
```

Comment:
Can we verify something about the client? Like it is pointing to the correct endpoint?

Reply:
It doesn't have any other property than `HttpPipeline Pipeline`